app/static/js/createEvents.js

```diff
@@ -384,6 +384,120 @@ function formatDate(originalDate) {
+  let trainingStatus = $("#checkIsTraining").is(":checked")
+  let serviceHourStatus = $("#checkServiceHours").is(":checked")
+  let engagementStatus = $("#checkEngagement").is(":checked")
+  let bonnersStatus = $("#checkBonners").is(":checked")
+  let allFieldFilled = true; //if there is text = true]
+  let seriesWeeklyId = $("#checkIsRepeating").is(":checked")
+  let isAllVolunteer = $("#pageTitle").text() == 'Create All Volunteer Training'
+
+  // .all is a class that groups the required filled present in all tempplate (event location and description)
+   $(".all").each(function(){
+    if (!$(this).is(":visible") || $(this).is(":disabled")) return; //check for hidden fields and skip them
+    if ($(this).val() === ""){ 
+      this.setCustomValidity("Please fill out the required field"); // do these actions
+      this.reportValidity();
+      allFieldFilled = false;
+    } else {
+       this.setCustomValidity(""); 
+       } 
+   });
+  
+  if (seriesEvent) {   
+    // .series is a class that groups the required filled present when the  series event toggle is toggled (event start date, end date and name)
+      $(".series").each(function(){
+      if (!$(this).is(":visible") || $(this).is(":disabled")) return;
+            // Skip event type checkboxes from regular validation
+      let elementId = $(this).prop("id");
+      if (elementId === "checkIsTraining" || elementId === "checkServiceHours" || //checking if any of the toggles are toggled
+          elementId === "checkEngagement" || elementId === "checkBonners") {
+        return; // Skip these, they'll be validated separately
+      }
+
+      if ($(this).val() === ""){
+        this.setCustomValidity("Please fill out the required field"); 
+        this.reportValidity();
+        allFieldFilled = false;
+      } else {
+        this.setCustomValidity(""); 
+        }
+    });
+  
+        if (seriesWeeklyId) {
+            $(".seriesWeekly").each(function(){
+            if (!$(this).is(":visible") || $(this).is(":disabled")) return;
+            if ($(this).val() === ""){
+              this.setCustomValidity("Please fill out the required field"); // do these actions
+              this.reportValidity();
+              allFieldFilled = false;
+            } else {
+              this.setCustomValidity(""); 
+              }
+            });
+        }   
+        if (!(trainingStatus || serviceHourStatus || engagementStatus || bonnersStatus)) {
+              document.getElementById("checkEngagement").setCustomValidity("Please select at least one of the event options.");
+              document.getElementById("checkEngagement").reportValidity();
+              allFieldFilled = false;
+            } else {
+              document.getElementById("checkEngagement").setCustomValidity("");
+          }
+      } 
+      else if (isAllVolunteer) {
+      $(".allV").each(function(){
+        if (!$(this).is(":visible") || $(this).is(":disabled")) return;
```

Comment:
The validity tag should disappear when the value is actively being entered in the datepicker, the event name, the description, the location, and the toggles. It may also be on one of the lines below, possibly, reportValidity, line 454. For example, once they start entering the description, the tag should disappear.